x/gov/types/v1/genesis.go

```diff
@@ -27,12 +28,63 @@ func (data GenesisState) Empty() bool {
 	return data.StartingProposalId == 0 || data.Params == nil
 }
 
-// ValidateGenesis checks if parameters are within valid ranges
+// ValidateGenesis checks if gov genesis state is valid ranges
+// It checks if params are in valid ranges
+// It also makes sure that the provided proposal IDs are unique and
+// that there are no duplicate deposit or vote records and no vote or deposits for non-existent proposals
 func ValidateGenesis(data *GenesisState) error {
 	if data.StartingProposalId == 0 {
 		return errors.New("starting proposal id must be greater than 0")
 	}
 
+	// verify proposal IDs
```

Comment:
We can be more precise with these comments given that we are really just checking for duplicates

`// Weed out duplicate proposals`

as verify is a more loaded term